app/models/section.rb

```diff
@@ -1,6 +1,36 @@
 class Section < ActiveRecord::Base
 	has_many 	:subscriptions
 	belongs_to	:user
+  has_many :articles, through: :subscriptions
 
 	validates	:title, uniqueness: { scope: :user }
+
+
+  def stories
+    # Get all the articles
+    # Check article similarity with other articles
+    # Group by similarity
+    cluster_articles(articles)
+  end
+
+  private
+  def cluster_articles(articles)
+    # p articles.length
+    # p articles
+    articles = articles.map { |x| x }
```

Comment:
Good to know that this is how to change ..CollectionProxy objects back to normal arrays.